tinytodo-go/internal/app/server/entitystore/entityuid/entityuid.go

```diff
@@ -0,0 +1,123 @@
+package entityuid
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/cedar-policy/cedar-examples/tinytodo-go/internal/app/server/entitystore/entitytype"
+	"github.com/cedar-policy/cedar-go/types"
+	"strings"
+)
+
+// EntityUID is a transparent wrapper around types.EntityUID used to represent entities in EntityStore, so that we
+// can define our own JSON marshaller.
+//
+// See EntityUID.MarshalJSON to understand why we need to define our own marshaller.
+type EntityUID struct {
+	types.EntityUID
+}
+
+// New creates an EntityUID from an entitytype.EntityType and ID.
+//
+// It is a simple wrapper around types.NewEntityUID with constraints on the valid types.
+func New(typ entitytype.EntityType, id string) EntityUID {
+	return EntityUID{
+		EntityUID: types.NewEntityUID(
+			types.EntityType(typ.String()),
+			types.String(id),
+		),
+	}
+}
+
+// ParseEntityUID converts the Cedar language representation of a types.EntityType into an EntityUID.
+// Additionally, it checks that the type of the EntityUID matches one of the enums defined in entitytype.EntityType.
+//
+// Example input:
+//
+//	"User::\"kesha\""
+func ParseEntityUID(uid string) (EntityUID, error) {
```

Comment:
This function name could probably just be `Parse`, given that it's already in `entityuid` package.

Reply:
fixed in 003a5f4633414efc9bf8a4abc27ebe4737834286

Reply:
On second thought, should `ParseEntityUID` be named `ParseCedarLanguageRepresentation` (or `ParseCedarRepresentation`) instead? This would be more explicit.

Reply:
Or perhaps `NewFromCedarLanguageRepresentation` (or `NewFromCedarRepresentation`)